zfsbootmenu/hook/zfsbootmenu-parse-commandline.sh

```diff
@@ -231,11 +231,18 @@ case "${root}" in
   ;;
 esac
 
-# Pool preference ending in ! indicates a hard requirement
-bpool="${root%\!}"
-if [ "${bpool}" != "${root}" ]; then
+# pool! : this pool must be imported before all others
+# pool!!: this pool, and no others, must be imported
+
+bpool="${root//[\!]/}"
+if [ "${root:(-1)}" = "!" ]; then
   # shellcheck disable=SC2034
   zbm_require_bpool=1
+
+  if [ "${root:(-2):1}" = "!" ]; then
+    # shellcheck disable=SC2034
+    zbm_only_bpool=1
+  fi
```

Comment:
What about (it seems "!" doesn't need to be escaped in these contexts)
```sh
# pool! : this pool must be imported before all others
# pool!!: this pool, and no others, must be imported

case "${root}" in
  *!!)
    zbm_require_bpool=yes
    root="${root%!!}"
    ;;
  *!)
    zbm_require_bpool=only
    root="${root%!}"
    ;;
  *)
    zbm_require_bpool=""
    ;;
esac
```
...and corresponding changes to preinit and init.

Reply:
Agreed, that's a lot easier to quickly read and understand what is intended.